var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -184,6 +184,16 @@ class Qt(Package):
     # causing qt to fail in ci.  This increases that limit to 1024.
     patch("qt59-qtbase-qtconfig256.patch", working_dir="qtbase", when="@5.9:5")
 
+    # with gcc@14: RapidJSON fails to build
+    # https://github.com/Tencent/rapidjson/issues/2277
+    # https://github.com/Tencent/rapidjson/pull/719
+    patch(
+        "https://patch-diff.githubusercontent.com/raw/Tencent/rapidjson/pull/719.patch?full_index=1",
+        sha256="ce341a69d6c17852fddd5469b6aabe995fd5e3830379c12746a18c3ae858e0e1",
+        working_dir="qtlocation/src/3rdparty/mapbox-gl-native/deps/rapidjson/1.1.0",
+        when="@5: %gcc@14:",
```

Comment:
Does this patch apply cleanly for all versions @5:?

I'd also recommend just applying the patch for all gcc version, unless it's known to be broken for older gcc versions. That will make sure that new versions will be tested with the patch, not overlooked because someone is testing with a different compiler.

Reply:
```suggestion
        when="@5.9.2: %gcc@14:",
```